### PR TITLE
Add xerces-c support

### PR DESCRIPTION
**Pull request by abellgithub**

### CODE REVIEW

**Comment by Ikke** [comment]
Hello @abellgithub, thanks for your pull request.

Could you update the commit message to match the aports style? For this commit, it would be:

> testing/xerces-c: new aport

See the [contributing guidelines][0] for more information.

Thanks!

[0]:https://github.com/alpinelinux/aports/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request-pr

File: testing/xerces-c/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    linux-headers
+    "
+install=""
+subpackages="$pkgname-doc"
+_pkgver=${pkgver//./_}
+_srcname="Xerces-C_${_pkgver}"
+source="https://github.com/apache/xerces-c/archive/${_srcname}.tar.gz"
+builddir="$srcdir/$pkgname-$_srcname"
+
+build() {
```
Comment (Ikke) [inline testing/xerces-c/APKBUILD]: If you add a prepare function like this:

```sh
prepare() {
    mkdir -p "$builddir"
    
    default_prepare
}
```
Then you can set `builddir` to `srcdir/$pkgname-$_srcname/build` and can remove the `mkdir build` and `cd` statements from the rest of the functions.

File: testing/xerces-c/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+pkgrel=0
+pkgdesc="Xerces-C++ is a validating XML parser written in a portable subset of C++."
+url="https://xerces.apache.org/index.html"
+arch="x86 x86_64"
+license="Apache-2.0"
+makedepends="
+    cmake
+    linux-headers
+    "
+install=""
```
Comment (Ikke) [inline testing/xerces-c/APKBUILD]: Empty global variables can be removed.

File: testing/xerces-c/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Contributor: Andrew Bell <andrew.bell.ia@gmail.com>
+# Maintainer: Andrew Bell <andrew.bell.ia@gmail.com>
+pkgname=xerces-c
+pkgver=3.2.2
+pkgrel=0
+pkgdesc="Xerces-C++ is a validating XML parser written in a portable subset of C++."
```
Comment (Ikke) [inline testing/xerces-c/APKBUILD]: You can leave out the project name from the description, so something like

```suggestion
pkgdesc="A validating XML parser written in a portable subset of C++."
```

File: testing/xerces-c/APKBUILD (hunk 1 above)
Comment (tcely) [reply testing/xerces-c/APKBUILD]: That approach is going to cause patching to fail.

Comment (Ikke) [reply testing/xerces-c/APKBUILD]: Fine for now.

**Pull request closed without merging**